api/src/services/survey-service.ts

```diff
@@ -816,4 +816,23 @@ export class SurveyService extends DBService {
   async deleteOccurrenceSubmission(submissionId: number): Promise<number> {
     return this.surveyRepository.deleteOccurrenceSubmission(submissionId);
   }
+
+  async getSurveyHasUnpublishedContent(surveyId: number): Promise<boolean> {
```

Comment:
nit: The name of this function is a bit hard to parse, I'd go for something like:
`doesSurveyHaveUnpublishedContent`
or the inverse
`isAllSurveyContentPublished`

And then maybe add some JSDoc, explain what the `true` vs `false` response means

Reply:
Same comment for the similar function in the project service.

Reply:
renamed and added jsdocs